Sources/SwiftKafka/KafkaClient.swift

```diff
@@ -56,6 +56,43 @@ final class KafkaClient {
         return rd_kafka_poll(self.kafkaHandle, timeout)
     }
 
+    /// Request a new message from the Kafka cluster.
+    ///
+    /// This method blocks for a maximum of `timeout` milliseconds.
+    ///
+    /// - Important: This method should only be invoked from ``KafkaConsumer``.
+    ///
+    /// - Parameter timeout: Maximum amount of milliseconds this method waits for a new message.
+    /// - Returns: A ``KafkaConsumerMessage`` or `nil` if there are no new messages.
+    /// - Throws: A ``KafkaError`` if the received message is an error message or malformed.
+    func consumerPoll(timeout: Int32) async throws -> KafkaConsumerMessage? {
```

Comment:
We should remove the `timeout` here otherwise people might actually block. Additionally, we should remove the `async` from this method. I don't see a reason why we are creating a continuation here. We should be able to just call `rd_kafka_consumer_poll` and bridge the message into a Swift type, destroy the pointer and return.

Reply:
Totally makes sense!